File: admin/server/router/ddz/room_sublevel.go
package ddz

import (
        v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
        "github.com/flipped-aurora/gin-vue-admin/server/middleware"
        "github.com/gin-gonic/gin"
)

type DDZRoomSublevelRouter struct{}

// InitDDZRoomSublevelRouter 初始化子分区路由
func (r *DDZRoomSublevelRouter) InitDDZRoomSublevelRouter(Router *gin.RouterGroup) {
        ddzRoomSublevelApi := v1.ApiGroupApp.DDZApiGroup.DDZRoomSublevelApi
        {
                // 需要权限的路由
                ddzRoomSublevelRouter := Router.Group("roomSublevel").Use(middleware.OperationRecord())
                ddzRoomSublevelRouter.Use(middleware.JWTAuth())
                {
                        ddzRoomSublevelRouter.GET("list", ddzRoomSublevelApi.GetRoomSublevelList)             // 获取子分区列表
                        ddzRoomSublevelRouter.GET("detail", ddzRoomSublevelApi.GetRoomSublevelByID)           // 获取子分区详情
                        ddzRoomSublevelRouter.POST("create", ddzRoomSublevelApi.CreateRoomSublevel)           // 创建子分区
                        ddzRoomSublevelRouter.PUT("update", ddzRoomSublevelApi.UpdateRoomSublevel)            // 更新子分区
                        ddzRoomSublevelRouter.DELETE("delete", ddzRoomSublevelApi.DeleteRoomSublevel)         // 删除子分区
                        ddzRoomSublevelRouter.POST("batchCreate", ddzRoomSublevelApi.BatchCreateDefaultSublevels) // 批量创建默认子分区
                        ddzRoomSublevelRouter.GET("byRoom", ddzRoomSublevelApi.GetRoomSublevelsByRoomConfig)  // 根据房间配置ID获取子分区
                        ddzRoomSublevelRouter.POST("refreshCache", ddzRoomSublevelApi.RefreshSublevelCache)   // 刷新缓存
                }
        }
}

File: admin/web/src/api/ddz/roomSublevel.js
import adminService from '@/utils/adminRequest'

// ==================== 子分区管理 ====================

// 获取子分区列表
export const getRoomSublevelList = (params) => {
  return adminService({
    url: '/ddz/roomSublevel/list',
    method: 'get',
    params
  })
}

// 获取子分区详情
export const getRoomSublevelDetail = (id) => {
  return adminService({
    url: '/ddz/roomSublevel/detail',
    method: 'get',
    params: { ID: id }
  })
}

// 创建子分区
export const createRoomSublevel = (data) => {
  return adminService({
    url: '/ddz/roomSublevel/create',
    method: 'post',
    data
  })
}

// 更新子分区
export const updateRoomSublevel = (data) => {
  return adminService({
    url: '/ddz/roomSublevel/update',
    method: 'put',
    data
  })
}

// 删除子分区
export const deleteRoomSublevel = (id) => {
  return adminService({
    url: '/ddz/roomSublevel/delete',
    method: 'delete',
    data: { ID: id }
  })
}

// 批量创建默认子分区
export const batchCreateDefaultSublevels = (roomConfigId) => {
  return adminService({
    url: '/ddz/roomSublevel/batchCreate',
    method: 'post',
    data: { roomConfigId }
  })
}

// 根据房间配置ID获取子分区列表
export const getSublevelsByRoomConfig = (roomConfigId) => {
  return adminService({
    url: '/ddz/roomSublevel/byRoom',
    method: 'get',
    params: { roomConfigId }
  })
}

// 刷新子分区缓存
export const refreshSublevelCache = (roomConfigId) => {
  return adminService({
    url: '/ddz/roomSublevel/refreshCache',
    method: 'post',
    params: { roomConfigId }
  })
}
